runtime/executor/tensor_parser_exec_aten.cpp

```diff
@@ -146,76 +185,43 @@ ET_NODISCARD Result<void*> getTensorDataPtr(
         s_tensor->extra_tensor_info()->fully_qualified_name() != nullptr,
         InvalidExternalData,
         "Fully qualified name of external tensor is null");
-    // Look up tensor in named data map.
-    Result<const TensorLayout> tensor_layout_res = named_data_map->get_metadata(
-        s_tensor->extra_tensor_info()->fully_qualified_name()->c_str());
-    if (!tensor_layout_res.ok()) {
-      return tensor_layout_res.error();
-    }
-    const TensorLayout& tensor_layout = tensor_layout_res.get();
-
-    // Compatibility checking.
-    ET_CHECK_OR_RETURN_ERROR(
-        static_cast<ScalarType>(s_tensor->scalar_type()) ==
-            tensor_layout.scalar_type(),
-        InvalidExternalData,
-        "Scalar type mismatch. Expected %hhd, got %hhd.",
-        static_cast<int8_t>(s_tensor->scalar_type()),
-        static_cast<int8_t>(tensor_layout.scalar_type()));
-    ET_CHECK_OR_RETURN_ERROR(
-        nbytes == tensor_layout.nbytes(),
-        InvalidExternalData,
-        "Nbytes mismatch. Expected %zu, got %zu.",
-        nbytes,
-        tensor_layout.nbytes());
-    int dim = s_tensor->sizes()->size();
-    ET_CHECK_OR_RETURN_ERROR(
-        dim == tensor_layout.sizes().size(),
-        InvalidExternalData,
-        "Dim mismatch. Expected %d, got %zu.",
-        dim,
-        tensor_layout.sizes().size());
-    for (int i = 0; i < dim; i++) {
-      ET_CHECK_OR_RETURN_ERROR(
-          s_tensor->sizes()->Get(i) == tensor_layout.sizes()[i],
-          InvalidExternalData,
-          "Sizes mismatch. Expected %d, got %d for size at index %d.",
-          s_tensor->sizes()->Get(i),
-          tensor_layout.sizes()[i],
-          i);
-      ET_CHECK_OR_RETURN_ERROR(
-          s_tensor->dim_order()->Get(i) == tensor_layout.dim_order()[i],
-          InvalidExternalData,
-          "Dim order mismatch. Expected %d, got %d for dim at index %d.",
-          s_tensor->dim_order()->Get(i),
-          tensor_layout.dim_order()[i],
-          i);
-    }
+    const char* key =
```

Comment:
Since this has a fairly long scope, `fqn` or `fully_qualified_name` would make it more self-documenting. It happens to be used as a key, but an FQN is what it _is_.